src/controller/CHIPDeviceController.cpp

```diff
@@ -2311,6 +2312,10 @@ CHIP_ERROR DeviceCommissioner::ParseICDInfo(ReadCommissioningInfo & info)
     }
 
     ReturnErrorOnFailure(err);
+    if (!isICD)
+    {
+        return CHIP_NO_ERROR;
```

Comment:
Why is this early return ok?  It'll possibly leave various pre-existing state on info.icd that should be cleared, no?  Like the userActiveModeTriggerHint and userActiveModeTriggerInstruction.

Reply:
yes, not ok, just move it to the location before retrieving IdleModeDuration